### ISSUE TITLE
shared functionality for all predictor data preparation

### ISSUE THREAD

**Issue by krsnapaudel** [BigDataWUR/AgML-CY-Bench#56]
Something like below would be useful. The pseudocode is meant as a rough guide.

```
# extract data for selected country
def extractForShape(predictor_dataset, shapefile, country):
    pass

# apply crop mask. This could involve resampling crop mask to predictor data resolution.
def maskByCrop(predictor_dataset, crop_mask):
   pass

# map pixels to admin regions
def mapToAdminLevel(predictor_dataset, admin_level):
   pass

# aggregate to admin regions
def aggregateToAdminLevel(predictor_dataset):
   pass

# usage for specific predictor data
def main():
    crop_name = "Grain maize"
    country = "FR"
    admin_level = "NUTS3"
    shapefile = <EU GISCO> # need to change it to filename to download
    crop_mask = <geoglam> # need to set it to mask based on crop

    data_source = "FPAR_500m"
    data_dir = "data/FPAR_500m"
    # download (done separately for each predictor dataset

    predictor_data = loadData(data_dir)
    unmasked_data = extractForShape(predictor_data, shapefile, "FR")
    masked_data = maskByCrop(unmasked_data, crop_mask)
    mapped_data = mapToAdminLevel(masked_data, admin_level)
    # Save the mapped data at original resolution
    aggr_data = aggregateToAdminLevel(masked_data)
    # Save the data at admin level
```

**Comment by krsnapaudel** [comment]
@AbdelrahmanAmr3 Some of your code for FPAR500 and AgERA5 could be factored and placed in a python script that can be shared by all predictor data preparation code. This is a rough guide. Feel free to reuse what you have and create some kind of shared API. Thanks.

**Comment by krsnapaudel** [comment]
Addressed by #199
